docs/ml-statistics.md

```diff
@@ -109,7 +109,8 @@ Refer to the [`ChiSquareTest` Python docs](api/python/index.html#pyspark.ml.stat
 ## Summarizer
 
 We provide vector column summary statistics for `Dataframe` through `Summarizer`.
-Available metrics are the column-wise max, min, mean, variance, and number of nonzeros, as well as the total count.
+Available metrics are the column-wise max, min, mean, sum, variance, std, squared sum, and number of nonzeros,
```

Comment:
I can see providing the sum maybe, but sum of squares (not squared sum right?)? is that useful? I know it's _available_ as a statistic, but i think the question is what will people expect to see here as compared to say https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.describe.html This doesn't provide sum although it provides percentiles, but I don't know if we should compute those.

Reply:
I can also see providing stddev instead of variance, but now we'd provide both, which is a little redundant, and we don't want to take away variance as it would unnecessarily break backwards compatibility. I don't know if the sqrt is such a big deal to compute.

Reply:
This PR is mainly for convenience:
1, `sum`: existing summarizer does not provide sum of weights, so if I want to compute `sum` in NaiveBayes, I need to use two udafs like `.agg(sum(w).as("weightSum"), Summarizer.metrics("mean", "count").summary(validateUDF(col($(featuresCol))), w))`, and then I need to call `BLAS.scal(weightSum, mean)`
2, `std`: I noticed that `std` is more widely used than `variance`;
3, `weightSum`: it is provided in `mllib.MultivariateOnlineSummarizer`, I donot know why `Summarizer` do not expose it.

In general I guess that the exisintg OPs of Vectors are not enough, if we can provide new OPs like `sqrt`,`square`,`abs` (like `np.sqrt`,`np.square`,`np.abs` in numpy), then we do not need to add above `std`,`sumL2`.

Reply:
sum and std I can understand. Maybe not weightSum? not sure.
What new ops are you imagining here as an alternative? You can already take the sqrt of variance without any new ops.

Reply:
Ok, I will update the metrics to only add `std` and `sum`.

> What new ops are you imagining here as an alternative? 

I guess we may add ops `sqrt`, `square` in `ml.Vector`.